tomee/tomee-security/src/main/java/org/apache/tomee/security/http/SavedRequest.java

```diff
@@ -102,12 +102,20 @@ public Enumeration<String> getHeaderNames() {
 
             @Override
             public Enumeration<String> getHeaders(String name) {
-                return Collections.enumeration(headers.get(name));
+                List<String> header = headers.get(name);
+                if (header == null) {
+                    header = Collections.emptyList();
+                }
+                return Collections.enumeration(header);
             }
 
             @Override
             public String getHeader(String name) {
-                return headers.get(name).get(0);
+                List<String> header = headers.get(name);
```

Comment:
in `getHeaders(String)` you assume `headers`may be null, this check is missing here